[RemoveDIs] Handle DPValues in LowerDbgDeclare

OCHyams opened the pull request:
I'm not sure what the best way to do stacked reviews is, but this is meant to only be a review for the last commit.

Based on top of https://github.com/llvm/llvm-project/pull/73500

OCHyams committed:
[RemoveDIs] Handle DPValues in LowerDbgDeclare

jmorse reviewed (approved):
LGTM with minor test nit, very much enjoying the auto-lambda situation.

OCHyams commented on llvm/test/Transforms/InstCombine/lifetime.ll:
```diff
@@ -1,5 +1,5 @@
 ; NOTE: Assertions have been autogenerated by utils/update_test_checks.py
-; RUN: opt < %s -passes=instcombine -S | FileCheck %s
+; RUN: opt --try-experimental-debuginfo-iterators < %s -passes=instcombine -S | FileCheck %s
```
Ah, that was an accident. Good catch, thanks.

OCHyams committed:
Replace original run line in test

OCHyams commented:
Reapplied original test RUN line, rebased patches, fixed conflicts by undoing changes in
```
llvm/test/Transforms/InstCombine/lower-dbg-declare.ll
llvm/test/Transforms/InstCombine/debuginfo.ll
llvm/test/Transforms/InstCombine/debuginfo-skip.ll
llvm/test/Transforms/InstCombine/dbg-scalable-store-fixed-frag.ll
llvm/test/DebugInfo/Generic/volatile-alloca.ll
```
- these tests now already have `--try-experiemtnal-debuginfo-iterators` in main.
